cupy/core/_routines_math.pyx

```diff
@@ -73,11 +73,15 @@ cdef ndarray _ndarray_prod(ndarray self, axis, dtype, out, keepdims):
 
 
 cdef ndarray _ndarray_sum(ndarray self, axis, dtype, out, keepdims):
+    cdef tuple reduce_axis, out_axis
     if cupy.cuda.cub_enabled:
-        if cub.can_use_device_reduce(cub.CUPY_CUB_SUM, self.dtype, self.ndim,
-                                     axis, dtype):
-            return cub.device_reduce(self, cub.CUPY_CUB_SUM, out=out,
-                                     keepdims=keepdims)
+        # if import at the top level, a segfault would happen when import cupy!
+        from cupy.core._kernel import _get_axis
```

Comment:
Have you tried moving this import to the top? My theory for the segfault is that `cupy.cuda` needs to be initialized before `cupy.core`, but importing `_get_axis` at the top of `cupy.cuda.cub` reverses the order. Now that the logic is moved to `cupy.core.*`, I suspect the segfault won't happen.

Same applies to all such import occurrences.

Reply:
This import is no longer done in these _routines_* files after the refactoring in f8a3db2